bindings/bindings.py

```diff
@@ -0,0 +1,558 @@
+            return f'"{out}"'
+        return out
+
+    def need_parse(self) -> bool:
+        return self.values.need_parse()
+
+    def load(self, val: Code) -> Code:
+        if not self.need_parse():
+            return val
+        return f"{{k: {self.values.load('v')} for k, v in {val}.items()}}"
+
+    def dump(self, val: Code) -> Code:
+        if not self.need_parse():
+            return val
+        return f"{{k: {self.values.dump('v')} for k, v in {val}.items()}}"
+
+
+class Sequence(TypeAnno):
+    def __init__(self, items):
+        self.items = items
+
+    def __repr__(self):
+        return f"Sequence[{self.items}]"
+
+    def annotation(self, prequoted=False) -> Code:
+        out = f"typing.Sequence[{self.items.annotation(True)}]"
+        if not prequoted:
+            return f'"{out}"'
+        return out
+
+    def need_parse(self) -> bool:
+        return self.items.need_parse()
+
+    def load(self, val: Code) -> Code:
+        if not self.need_parse():
+            return val
+        return f"[{self.items.load('x')} for x in {val}]"
+
+    def dump(self, val: Code) -> Code:
+        if not self.need_parse():
+            return val
+        return f"[{self.items.dump('x')} for x in {val}]"
+
+
+class Class(TypeDef):
+    def __init__(self, name: str, members: typing.Dict[str, TypeAnno]):
+        self.name = name
+        self.members = members
+
+    def load(self, val: Code) -> Code:
+        return f"{val}.to_json()"
+
+    def dump(self, val: Code) -> Code:
+        return f"{self.name}.from_json({val})"
+
+    def gen_def(self) -> Code:
+        out = [f"class {self.name}:"]
+        out += ["    def __init__("]
+        out += ["        self,"]
+        out += [f"        {k}: {v.annotation()}," for k, v in self.members.items()]
+        out += ["    ):"]
+        out += [f"        {k} = {k}" for k in self.members]
+        out += [""]
+        out += ["    @classmethod"]
+        out += ["    def from_json(cls, obj):"]
+        out += ["        return cls("]
+        for k, v in self.members.items():
+            if v.need_parse():
+                parsed = v.load(f'obj["{k}"]')
+            else:
+                parsed = f'obj["{k}"]'
+            out.append(f"""             {k}={parsed},""")
+        out += ["        )"]
+        out += [""]
+        out += ["    def to_json(self):"]
+        out += ["        return {"]
+        for k, v in self.members.items():
+            if v.need_parse():
+                parsed = v.dump(f"self.{k}")
+            else:
+                parsed = f"self.{k}"
+            out.append(f'             "{k}": {parsed},')
+        out += ["        }"]
+
+        return "\n".join(out)
+
+
+class Enum(TypeDef):
+    def __init__(self, name, members):
+        self.name = name
+        self.members = members
+
+    def load(self, val: Code) -> Code:
+        return f"{val}.value"
+
+    def dump(self, val: Code) -> Code:
+        return f"{self.name}({val})"
```

Comment:
I think these two are swapped

```suggestion
    def load(self, val: Code) -> Code:
        return f"{self.name}({val})"

    def dump(self, val: Code) -> Code:
        return f"{val}.value"
```